src/backend/gmail.js

```diff
@@ -181,72 +181,135 @@ function addEmailToSchedule(subject, date) {
+  var url = 'https://www.googleapis.com/gmail/v1/users/me/drafts/send'
+  var headers = {
+    Authorization: 'Bearer ' + ScriptApp.getOAuthToken()
+  };
+  var params = {
+    method: "post",
+    contentType: "application/json",
+    headers: headers,
+    muteHttpExceptions: true,
+    payload: JSON.stringify(draftMsg)
+  };
+  var check = UrlFetchApp.getRequest(url, params)
+  var response = UrlFetchApp.fetch(url, params);
+
+  var result = response.getResponseCode();
+  if (result == '200') {  // OK
+    return "Delivered";
+    //return JSON.parse(response.getContentText());
+  }
+  else {
+    // This is only needed when muteHttpExceptions == true
+    var err = JSON.parse(response.getContentText());
+    throw new Error( 'Error (' + result + ") " + err.error.message );
+  }
+}
 
-            GmailApp.sendEmail(message.getTo(), message.getSubject(), body, options);
-            message.moveToTrash();
-            return "Delivered";
-        } else {
-            return "Message not found in Drafts";
-        }
-    } catch (e) {
-        return e.toString();
+/**
+ * https://stackoverflow.com/questions/27206595/how-to-send-a-draft-email-using-google-apps-script
+ * Gets the draft message content that corresponds to a given Gmail Message ID.
+ * Throws if unsuccessful.
+ * See https://developers.google.com/gmail/api/v1/reference/users/drafts/get.
+ *
+ * @param {String}     messageId   Immutable Gmail Message ID to search for
+ * @param {String}     optFormat   Optional format; "object" (default) or "json"
+ *
+ * @returns {Object or String}     If successful, returns a Users.drafts resource.
+ */
+function getDraftMsg( messageId, optFormat ) {
+  var draftId = getDraftId( messageId );
+
+  var url = 'https://www.googleapis.com/gmail/v1/users/me/drafts'+"/"+draftId;
+  var headers = {
+    Authorization: 'Bearer ' + ScriptApp.getOAuthToken()
+  };
+  var params = {
+    headers: headers,
+    muteHttpExceptions: true
+  };
+  var check = UrlFetchApp.getRequest(url, params)
+  var response = UrlFetchApp.fetch(url, params);
+
+  var result = response.getResponseCode();
+  if (result == '200') {  // OK
+    if (optFormat && optFormat == "JSON") {
+      return response.getContentText();
+    }
+    else {
+      return JSON.parse(response.getContentText());
     }
+  }
+  else {
+    // This is only needed when muteHttpExceptions == true
+    var error = JSON.parse(response.getContentText());
+    throw new Error( 'Error (' + result + ") " + error.message );
+  }
+}
+
+/**
+ * Gets the draft message ID that corresponds to a given Gmail Message ID.
+ *
+ * @param {String}     messageId   Immutable Gmail Message ID to search for
+ *
+ * @returns {String}               Immutable Gmail Draft ID, or null if not found
+ */
+function getDraftId( messageId ) {
+  if (messageId) {
+    var drafts = getDrafts();
+
+    for (var i=0; i<drafts.length; i++) {
+      if (drafts[i].message.id === messageId) {
```

Comment:
@ChristopherDavenport The getDrafts() function is implement in the PR and uses the gmail api (not the the one built into scripts).

@jcricket I didn't realize that the Script needed to be authorized in both of those places. After turning them on it worked fine but we'll still want the error checking for new users.
Do you know if every individual user has to turn on those settings or one who owns the script?
We'll definitely want to add a note about that in the setup instructions if it's required.
I guess at this point it would also be wise to add some type of versioning so people can be aware of major changes and be able to update their scripts later.

Also, I'm thinking this will fix issue #2 so that would be great.